smdebug/tensorflow/keras.py

```diff
@@ -418,6 +438,34 @@ def _save_tensor_to_file(self, tensor_name, tensor_value, collections):
                 collection.set_tensor_ref(tensor_ref)
             self._save_for_tensor(tensor_name, t, check_before_write=True)
 
+    def save_gradients_from_logs(self, gradients):
+        if gradients is not None:
+            if len(self.gradient_name_to_tensor_position_map) == 0 and self._gradients_set is False:
+                # Map has not yet been initialized
+                for index, grad_var in enumerate(zip(gradients, self.model.trainable_variables)):
+                    g, v = grad_var  # split gradients and trainable variables tuple
+                    layer_name = v.name
+                    if len(layer_name.split(":")) > 1:
```

Comment:
if the character `:` is not present in the string, then split returns a single single string.
The next line will simply be a redundant operation.

```
>>> s = "helloworld"
>>> s.split(':')
['helloworld']
```

Reply:
Do we even need this check? As pointed out in your example, if `:` is not present, it will return the string wrapped in an array. So it looks like either way, `layer_name.split(":")[0]` is enough to get the expected value.
```
>>> s1 = "helloworld"
>>> s1.split(':')[0]
'helloworld'
>>> s2 = "hello:world"
>>> s2.split(':')[0]
'hello'
```

Reply:
You're right. It seems to work fine for empty strings as well.